### PR TITLE
VizTooltips: Use global portal

### PR DESCRIPTION
**Pull request by leeoniya**
this refactors `newVizTooltips` to use a global portal for tooltips and annotation editor/infotip. this fixes a number of tricky z-index issues with portalling tooltips into the panels, especially with shared tooltips, but also with the panel resize-handle in each panel appearing on top of the tooltips.

this also removes the condition to use old tooltips for dashboards with shared tooltip mode. the new behavior is to always use Multi/All mode for tooltips in synced panels.

there will be a follow-up PR to revert previous z-index changes we did in dashboards and scenes that were needed for panel-local portalling:

1. descending z-index:
  - https://github.com/grafana/grafana/pull/75328
  - https://github.com/grafana/grafana/pull/76675
  - https://github.com/grafana/scenes/issues/366
2. boostZIndex (LayoutContext):
  - https://github.com/grafana/grafana/pull/80116
  - https://github.com/grafana/scenes/pull/530

we could potentially revert https://github.com/grafana/grafana/pull/75499, but i think that can stay in place.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
@leeoniya tested this and works great and now is above the resize handle :) 

And the performance for the shared tooltip mode seems really good  (all move very smoothly)

File: public/app/core/components/TimelineChart/timeline.ts [hunk 1]
```diff
@@ -384,27 +384,23 @@ export function getConfig(opts: TimelineCoreOptions) {
...
     if (cx < 0) {
       return;
     }
 
-    for (let i = 0; i < cys.length; i++) {
-      let cy2 = cys[i];
-
-      qt.get(cx, cy2, 1, 1, (o) => {
-        if (pointWithin(cx, cy2, o.x, o.y, o.x + o.w, o.y + o.h)) {
+    qt.get(cx, 0, uPlot.pxRatio, 1e4, (o) => {
```
Comment (codeincarnate) [inline public/app/core/components/TimelineChart/timeline.ts:395]: Would be nice to have a short comment explaining this chunk

**Comment by codeincarnate** [comment]
Getting an error with this enabled in certain conditions in which it appears to be causing some overflow causing the global scrollbar to flicker. See screen recording:

https://github.com/grafana/grafana/assets/199847/b5cecc2a-31a0-4545-a9d3-38746a6fab06

**Review by codeincarnate — CHANGES_REQUESTED** [review]
See comment with video :confounded:

File: packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx [hunk 2]
```diff
@@ -105,13 +109,16 @@ export const TooltipPlugin2 = ({
   queryZoom,
   maxWidth,
   maxHeight,
+  syncTooltip = () => false,
 }: TooltipPlugin2Props) => {
   const domRef = useRef<HTMLDivElement>(null);
+  const portalRoot = useRef<HTMLElement | null>(null);
 
-  const [{ plot, isHovering, isPinned, contents, style, dismiss }, setState] = useReducer(mergeState, INITIAL_STATE);
+  if (portalRoot.current == null) {
```
[the remark this answers is not in the document]
Comment (leeoniya) [reply packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx:117]: i prefer to use `== null` in all cases unless i need to discern between null and undefined, which is basically never.

**Comment by leeoniya** [comment]
> Getting an error with this enabled in certain conditions

what error are you getting?

the occasional flicker is known, but i want to leave the fix for later as it's not a 1-liner - i think we need to render the tooltip initially with `visibility: hidden`, then check viewport overlap and re-render again with `visibility: visible` after adjusting position...or something. not sure exactly yet but dont want to hold up this pr on it.

**Review by codeincarnate — APPROVED** [review]
No worries on the error then, it can be in a follow up PR

**Review by adela-almasan — APPROVED** [review]
⭐

**Pull request merged**